File: backend/src/main/java/info/gamewise/lor/videos/domain/LoRVideo.java
package info.gamewise.lor.videos.domain;

import com.github.brunosc.fetcher.domain.VideoThumbnails.ThumbnailItem;

import java.time.LocalDateTime;
import java.util.Set;

public record LoRVideo(String url,
                       String title,
                       String deckCode,
                       VideoChannel channel,
                       Set<VideoChampion> champions,
                       Set<VideoRegion> regions,
                       LocalDateTime publishedAt,
                       ThumbnailItem thumbnail) {

    @Override
    public String toString() {
        return "LoRVideo{" +
                "url='" + url + '\'' +
                ", title='" + title + '\'' +
                ", deckCode='" + deckCode + '\'' +
                ", channel=" + channel +
                '}';
    }
}

File: backend/src/main/kotlin/info/gamewise/lor/videos/domain/LoRVideo.kt
package info.gamewise.lor.videos.domain

import com.github.brunosc.fetcher.domain.VideoThumbnails.ThumbnailItem
import java.time.LocalDateTime

data class LoRVideo(val url: String,
                    val title: String,
                    val deckCode: String,
                    val channel: VideoChannel,
                    val champions: Set<VideoChampion>,
                    val regions: Set<VideoRegion>,
                    val publishedAt: LocalDateTime,
                    val thumbnail: ThumbnailItem)